igc_lib.py

```diff
@@ -71,7 +73,148 @@ def _rawtime_float_to_hms(timef):
 
     return hms((time/3600), (time%3600)/60, time%60)
 
-
+class Turnpoint:
+    """ single turnpoint in a task.
+    Attributes: 
+        lat: a float, latitude in degrees 
+        lon: a float, longitude in degrees
+        radius: radius of cylinder or line in km
+        kind: type of turnpoint; "start_exit", "start_enter", "cylinder", "End_of_speed_section", "goal_cylinder", "goal_line"
+        
+    """
+    def __init__(self, lat, lon, radius, kind):
+        self.lat = lat
+        self.lon = lon
+        self.radius = radius
+        self.kind = kind
+        assert kind in ["start_exit", "start_enter", "cylinder", "End_of_speed_section", "goal_cylinder", "goal_line"], \
+            "turnpoint type is not valid: %r" % kind
+        
+
+    def in_radius(self, fix):
+        """Checks whether the provided GNSSFix is within the radius"""
+        lat1, lon1, lat2, lon2 = map(math.radians, [self.lat, self.lon, fix.lat, fix.lon])
+        distance = EARTH_RADIUS_KM * _sphere_distance(lat1, lon1, lat2, lon2)
+        return distance < self.radius
+        
+class Task:
+    """Stores a task definition and checks if a flight has achieved the turnpoints in the task.
+    Attributes: 
+        turnpoints: A list of Turnpoint objects.
+        start_time: Raw time (seconds past midnight). The time the race starts. The pilots must start at or after this time.
+        end_time: Raw time (seconds past midnight). The time the race ends. The pilots must finish the race at or before this time. 
+                  No credit is given for distance covered after this time.
+    """
+    
+    @staticmethod
+    def create_from_lkt_file(filename):
+        """ Creates Task from LK8000 task file, which is in xml format.
+            LK8000 does not have ESS or task finish time.
+            For the goal, at the moment, Turnpoints can't handle goal cones or lines, for this reason we default to goal_cylinder.
+        """
+                               
+        # Open XML document using minidom parser
+        DOMTree = xml.dom.minidom.parse(filename)
+        task = DOMTree.documentElement
+
+        # Get the taskpoints, waypoints and time gate
+        taskpoints = task.getElementsByTagName("taskpoints")[0]
+        waypoints = task.getElementsByTagName("waypoints")[0]
+        gate = task.getElementsByTagName("time-gate")[0]
+        tpoints = taskpoints.getElementsByTagName("point")
+        wpoints = waypoints.getElementsByTagName("point")
+        start_time = gate.getAttribute("open-time")
+               
+        start_hours, start_minutes = start_time.split(':')
+        start_time = int(start_hours) * 3600 + int(start_minutes) * 60
+        end_time = 86399  #default end_time of 23:59
+        
+        #create a dictionary of names and a list of longitudes and latitudes as the waypoints co-ordinates are stored separate to turnpoint details
+        coords = defaultdict(list)
```

Comment:
I tried to avoid waypoints as this is already used in the xml file, and as a variable above. Looking at the code I can probably reuse waypoints. 

For the dictionary looking at the forums this seems to be the most popular way of making a dictionary of lists. Is there a better way?